imodqgis/widgets/pseudocolor_widget.py

```diff
@@ -252,11 +260,39 @@ def remove_selection(self):
         for item in self.table.selectedItems():
             self.table.takeTopLevelItem(self.table.indexOfTopLevelItem(item))
 
+    def _create_colorramp(self, boundaries, colors):
```

Comment:
I would move both _create_colorramp and most of load_classes into a separate function (e.g. load_and_create_colorramp), not associated with this class.

Then, load_classes would look like this:

```python
def load_classes(self):
    path, _ = QFileDialog.getOpenFileName(self, "Load colormap", "", "*.txt")
    colorramp, boundaries, colors = load_and_create_colorramp(path)
    self.set_color_items_in_table(boundaries, colors)
    self.color_ramp_button.setColorRamp(colorramp)
```

That way, all the class logic is nicely separated and load_and_create_colorramp is a more pure function.

Reply:
I wasn't really happy with a function which returned colorramp, as well as boundaries and colors; so I decided to place better comments better instead here, and reduce the code duplication when it came to unpacking values instead.